### PR TITLE
Fix missing generic_urlparse() in functions of urls.py

### PR DESCRIPTION
**Pull request by tux-00**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

lib/ansible/module_utils/urls.py

##### ANSIBLE VERSION

```
ansible 2.3.0 (devel fccb434d4e) last updated 2016/11/30 21:00:18 (GMT +200)
  lib/ansible/modules/core: (detached HEAD 6890003c4f) last updated 2016/11/30 23:01:22 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 20ea46642b) last updated 2016/11/30 23:01:22 (GMT +200)
  config file = 
  configured module search path = Default w/o overrides
```

##### SUMMARY

Fix "FIXME:" comments in lib/ansible/module_utils/urls.py and add missing quotation mark into an example.

```

```

### CODE REVIEW

**Comment by tux-00** [comment]
Ansible CI use pylint-1.6.5 and seems fail due to this issue: https://github.com/PyCQA/pylint/issues/1295

**Comment by gundalow** [comment]
Closing & reopening to trigger CI

**Comment by tux-00** [comment]
My code is not good. I need to commit another one to pass the CI but I need help from someone who know the code of urls.py to be sure.

File: lib/ansible/module_utils/urls.py [hunk 1]
```diff
@@ -792,9 +792,18 @@ def maybe_add_ssl_handler(url, validate_certs):
...
+        if ':' in netloc:
+            hostname, port = netloc.split(':', 1)
+            port = int(port)
+        else:
+            hostname = netloc
+            port = 443
         # create the SSL validation handler and
         # add it to the list of handlers
-        return SSLValidationHandler(parsed.get('hostname'), parsed.get('port'))
+        return SSLValidationHandler(hostname, port)
```
Comment (tux-00) [inline lib/ansible/module_utils/urls.py]: When I use `return SSLValidationHandler(parsed.get('hostname'), parsed.get('port'))` the test is failing.

**Comment by Akasurde** [comment]
@tux-00 Are you still working on this ? needs_info

**Comment by sivel** [comment]
This has already been resolved in https://github.com/ansible/ansible/pull/34372

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Pull request closed without merging**